src/app/payments/get-protocol-fee.ts

```diff
@@ -128,10 +136,15 @@ const estimateLightningFee = async ({
       return PartialResult.err(lndService)
     }
 
-    const routeResult = await lndService.findRouteForInvoice({
-      invoice,
-      amount: btcPaymentAmount,
-    })
+    const skipProbe =
+      intersect(parseFinalHopsFromInvoice(invoice), getPubkeysToSkipProbe()).length > 0
```

Comment:
I think this logic belongs inside the builder - it can take the pub keys to skip as config:
```
type LightningPaymentFlowBuilderConfig = {
  localNodeIds: Pubkey[]
  skipProbeKeys: Pubkey[]
  usdFromBtcMidPriceFn: UsdFromBtcMidPriceFn
  btcFromUsdMidPriceFn: BtcFromUsdMidPriceFn
}
```
or something like that.

This step:
```
type LPFBWithConversion<S extends WalletCurrency, R extends WalletCurrency> = {
  withRoute({
    pubkey,
    rawRoute,
  }: {
    pubkey: Pubkey
    rawRoute: RawRoute
  }): Promise<PaymentFlow<S, R> | ValidationError | DealerPriceServiceError>
  withoutRoute(): Promise<PaymentFlow<S, R> | ValidationError | DealerPriceServiceError>

  btcPaymentAmount(): Promise<BtcPaymentAmount | DealerPriceServiceError>
  usdPaymentAmount(): Promise<UsdPaymentAmount | DealerPriceServiceError>

  isIntraLedger(): Promise<boolean | DealerPriceServiceError>
}
```
Already has some query methods - could also add another one `shouldSkipProbe`?

There is no way this logic should be integration tested.

Reply:
Done: https://github.com/GaloyMoney/galoy/pull/1505/commits/1ba3743f334fdc45148762abe3312d6334558703